buildSrc/src/integTest/groovy/org/elasticsearch/gradle/YamlRestCompatTestPluginFuncTest.groovy

```diff
@@ -8,13 +8,25 @@
 
 package org.elasticsearch.gradle
 
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.ObjectReader
+import com.fasterxml.jackson.databind.ObjectWriter
+import com.fasterxml.jackson.databind.SequenceWriter
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
 import org.elasticsearch.gradle.fixtures.AbstractRestResourcesFuncTest
 import org.elasticsearch.gradle.internal.rest.compat.YamlRestCompatTestPlugin
 import org.gradle.testkit.runner.TaskOutcome
 
 class YamlRestCompatTestPluginFuncTest extends AbstractRestResourcesFuncTest {
```

Comment:
We should add a test case here to verify up-to-date checking as I described in my other comment. That is, configure a transform, run the build (success). Run the build again (up-to-date). Change some config in the transform (don't add/remove any transforms), run the build again (success).

Ensuring build correctness with this kind of thing is not easy, which is why all the back and forth. Builds that aren't correct (don't rebuild stuff when they should) make developers go crazy as they make a change in the IDE, run a build, and get the same result, leading them to thing they're going mad. It's a bit of a pain but we need to ensure everything is wired up just so.

Reply:
+1 ... I have all but last part ("Change some config in the transform (don't add/remove any transforms), run the build again (success)") https://github.com/elastic/elasticsearch/pull/68338/files#diff-3f396fefa7f5f573c7df27b6f9e019d2429e3483403149c6cddce256d823d942R310

Reply:
done as part of [`db82fd8` (#68338)](https://github.com/elastic/elasticsearch/pull/68338/commits/db82fd8671d4ce610f974838dbb59e25ee06b78d)